### PR TITLE
Generating update events when conversation access mode changes

### PR DESCRIPTION
**Pull request by vytis**
### Issues

To have real-world integration tests we need mock transport to generate update events when conversation access mode changes.

### Solutions

Observing  changes in conversations and producing an event when needed. Also added a convenience method `MockConversation.set(allowGuests:Bool)` to set the relevant values.

### CODE REVIEW

File: Source/MockTransportSession.swift [hunk 1]
```diff
@@ -67,6 +67,63 @@ public extension MockTransportSession {
...
+        return conversations
+            .flatMap { object -> MockConversation? in
+                object as? MockConversation
+            }.filter { conversation -> Bool in
+                conversation.type != .invalid && conversation.selfIdentifier == self.selfUser.identifier
+            }
+    }
+
+    @objc(pushEventsForInsertedConversations:updated:shouldSendEventsToSelfUser:)
+    public func pushEventsForConversations(inserted: Set<NSManagedObject>, updated: Set<NSManagedObject>, shouldSendEventsToSelfUser: Bool) -> [MockPushEvent] {
```
Comment (mikeger) [inline Source/MockTransportSession.swift]: Does `inserted` contain only `MockConversation` objects? Might be useful to specify the type then?

Comment (vytis) [reply Source/MockTransportSession.swift]: It contains all objects that were inserted after save. That's why there is initial filtering here:
https://github.com/wireapp/wire-ios-mocktransport/pull/74/files/220dd918e9fc1f2cd58f9c742620c97074ba0786#diff-4831fdba73321f46453f16e92dafff1aR73

Comment (mikeger) [reply Source/MockTransportSession.swift]: Ok, then the label is slightly confusing (says `conversations:`).

Comment (vytis) [reply Source/MockTransportSession.swift]: It is a bit, but basically there is a method for each type of objects we want to have update events. They are all on `MockTransportSession` and can pick and choose all objects they want:
https://github.com/wireapp/wire-ios-mocktransport/blob/develop/Source/MockTransportSession.m#L947

Comment (mikeger) [reply Source/MockTransportSession.swift]: Right, so not feasible to rename I guess.

**Pull request merged**